docs/guides/implementation_blueprint/data-synchronization.md

```diff
@@ -0,0 +1,200 @@
+
+The following endpoints can be used to synchronize customer data:
+
+- [List Customers](ref:list-customers)
+- [Get Customer](ref:get-customer)
+- [List Customer Activities](ref:list-customer-activities)
+- [List Customer's Segments](ref:list-customer-segments)
+- [Create Customer](ref:create-customer)
+- [Import and Update Customers using CSV](ref:import-customers-using-csv)
+- [Update Customer](ref:update-customer)
+- [Update Customer's Consents](ref:update-customers-consents)
+- [Update Customer’s consents (client-sde)](ref:update-customers-consents-client-side)
+- [Update Customers in bulk](ref:update-customers-in-bulk)
+- [Update Customers' Metadata in bulk](ref:update-customers-metadata-in-bulk)
+- [Delete Customer](ref:delete-customer)
+- [Delete Customer Permanently](ref:delete-customer-permanently)
+
+For reference, see also [the customer object](ref:customer-object) and [customer activity object](ref:customer-activity-object).
+
+<!-- do we also list all other endpoints that have a customer object that can upsert data, e.g. redeem stackable discount, check eligibility? -->
+
+## Product synchronization
+
+Products can be stored in Voucherify but it is not required. The product validation <!-- validation? --> takes place during the following actions:
+- Qualification
+- Validation
+- Redemption
+
+The product validation is based on the API payload and the data stored in Voucherify's product catalog. However, if a product does not exist in Voucherify, the validation will take into consideration the payload alone.
+
+> 🚧 Product data in the payload alone
+>
+> Unlike customer data, product data provided in the payload alone do not create a new product in Voucherify. Products need to be created or imported.
+
+The products in Voucherify can be grouped into static or dynamic collections based on their parameters. Collections can be used in validation rules to model product-specific limits and loyalty-earning rules.
+
+### Product synchronization options
+
+#### Importing products
+
+You can [import your product database](https://support.voucherify.io/article/515-products#import-products-skus "Import Products and SKUs by CSV") to Voucherify with a CSV file. This method can also be used to update the names of existing products.
+
+#### Creating and updating products via API
+
+You can create and update your products via the following API endpoints:
+- [Create product](ref:create-product)
+- [Update product](ref:update-product)
+
+#### Overriding product data
+
+The override set to `true` is used to store the product information in the system. If the product does not exist, it will be created with a `source_id`. If the product does exist, the provided values for the name, price, and metadata will replace those already stored in Voucherify.
+
+<!-- I don't get it; how does it work? Why is it in response? Also, in the HS article on products, we read 
```

Comment:
It's not in the response. It's in the request. If it's used in the redemption or create order request, the product is updated.

Reply:
Clarified further on Slack. Current docs are misleading in this regard 🙁